session/src/main/java/org/apache/iotdb/session/SessionDataSet.java

```diff
@@ -93,7 +93,12 @@ private RowRecord constructRowRecordFromValueArray() throws StatementExecutionEx
         switch (dataType) {
           case BOOLEAN:
             boolean booleanValue = BytesUtils.bytesToBool(valueBytes);
-            field.setBoolV(booleanValue);
+            if (org.apache.iotdb.rpc.Config.boolFormat == org.apache.iotdb.rpc.Config.Constant.BOOLEAN) {
+              field.setBoolV(booleanValue);
+            } else {
+              field = new Field(TSDataType.deserialize((short) 1));
+              field.setIntV(booleanValue ? 1 : 0);
+            }
```

Comment:
Hi, I think we should not change the type of Field. This issue only changes how the getString() method returns data.